File: src/main/java/cn/taskeren/minequery/config/MineQueryConfigModMenu.java
package cn.taskeren.minequery.config;

import com.terraformersmc.modmenu.api.ConfigScreenFactory;
import com.terraformersmc.modmenu.api.ModMenuApi;
import me.shedaniel.autoconfig.AutoConfig;

public class MineQueryConfigModMenu implements ModMenuApi {

	@Override
	public ConfigScreenFactory<?> getModConfigScreenFactory() {
		return screen -> AutoConfig.getConfigScreen(MineQueryConfig.class, screen).get();
	}
}

File: src/main/kotlin/cn/taskeren/minequery/config/MineQueryConfigModMenu.kt
package cn.taskeren.minequery.config

import com.terraformersmc.modmenu.api.ConfigScreenFactory
import com.terraformersmc.modmenu.api.ModMenuApi
import me.shedaniel.autoconfig.AutoConfig

class MineQueryConfigModMenu : ModMenuApi {

	override fun getModConfigScreenFactory(): ConfigScreenFactory<*> {
		return ConfigScreenFactory { parent ->
			AutoConfig.getConfigScreen(MineQueryConfig::class.java, parent).get()
		}
	}
}